octoprint_telegram/telegramCommands.py

```diff
@@ -294,7 +293,29 @@ def cmdSys(self,chat_id,from_id,cmd,parameter):
 					i += 1
 			if len(tmpKeys) > 0:
 				keys.append(tmpKeys)
-			keys.append([["Restart OctoPrint","/sys_sys_Restart OctoPrint"]])
+
+			tmpKeys = []
+			i = 1
+			serverCommands = { 'systemShutdownCommand':  ["Shutdown System","/sys_sys_Shutdown System"],
+ 					   'serverRestartCommand':   ["Restart OctoPrint", "/sys_sys_Restart OctoPrint"],
+ 				   'systemRestartCommand':   ["Reboot System", "/sys_sys_Reboot System"]
+			for index in serverCommands:
+				commandText = self.main._settings.global_get(['server', 'commands', index])
+				if commandText is not None:
+					tmpKeys.append(serverCommands[index])
+					if i % 2 == 0:
+						keys.append(tmpKeys)
+						tmpKeys = []
+					i += 1
+			if len(tmpKeys) > 0:
+				keys.append(tmpKeys)
+
+			if len(keys) > 0 :
+				message_text = " The following System Commands are known."
+			else:
+				message_text = " No known System Commands."
+			message = self.gEmo('info') + message_text
+
 			keys.append([["Reboot System","/sys_sys_Reboot System"],["Shutdown System","/sys_sys_Shutdown System"]])
```

Comment:
Doesn't this add the reboot and shutdown command, which were added in L299, again?

Reply:
You correct had to remove the two lines...